src/lib/support/PersistedCounter.h

```diff
@@ -110,6 +110,45 @@ class PersistedCounter : public MonotonicallyIncreasingCounter<T>
         return MonotonicallyIncreasingCounter<T>::Init(startValue);
     }
 
+    CHIP_ERROR SetValue(T value) override
+    {
+        VerifyOrReturnError(mStorage != nullptr, CHIP_ERROR_INCORRECT_STATE);
+        VerifyOrReturnError(mKey.IsInitialized(), CHIP_ERROR_INCORRECT_STATE);
+
+        // Calculate previous Epoch from value
+        T previousEpoch = value - (value % mEpoch);
```

Comment:
I'm not sure this makes sense.  Nothing ensures that the stored epoch values are multiples of mEpoch.  This `previousEpoch` value has nothing to do with anything "previous".

What we are really trying to compute here is a place to restart the counter next time, no?  We can just as easily use `value + mEpoch` (just like in `Init()` we use "value read from storage + mEpoch").  If we are doing something else, we need a clear explantion for why.

Reply:
Good point! I'll simplify the logic around this.